setup.py

```diff
@@ -50,7 +51,19 @@
 if os.environ.get("TEST_JUJU3"):
     install_require.append('juju')
 else:
-    install_require.append('juju<3.0.0')
+    try:
+        version = subprocess.check_output(['juju', '--version'],
+                                          universal_newlines=True).strip()
+        print('juju --version ->', version)
+        if int(version[0]) >= 3:
+            juju_ver_n = version[0:3]
+            juju_ver_n1 = "%s.%s" % (version[0], int(version[2]) + 1)
+            install_require.append('juju>=%s,<%s' % (juju_ver_n,
```

Comment:
I am sceptical about using `juju>=juju_ver_n` especially for cases of any new juju major series release (using terminology mentioned in [1]) like 3.x -> 3.y or 4.x -> 4,y
Since there is usually couple of days gap between Juju release and py-libjuju release.
Here is the past data for 3.6 release for juju and pylibjuju [2] [3]

We need to check with juju team if they really follows what they mentioned in documentation [1] to use `juju>=version[0],<juju_ver_n1`

[1] https://canonical-juju.readthedocs-hosted.com/en/latest/user/reference/juju/juju-cross-version-compatibility/#juju-cli-controllers-and-ref-agents
[2] https://github.com/juju/juju/releases/tag/v3.6.0
[3] https://github.com/juju/python-libjuju/releases/tag/3.6.0.0

Reply:
After thinking a bit, the code change seems fine since if pylibjuju is not present for the version, it will throw an error during installation which means the particular juju version cannot be used.

Reply:
I'd expect libjuju to significantly lag behind juju for new major versions. But then, running e.g. libjuju 3.x with juju 4.x is likely to run into compatibility issues, so I'm thinking failing the run for major version mismatches might be a good decision anyway.

Wonder if the version parsing should be made more robust to better handle version string format changes or double-digit versions though.

Reply:
I pushed a new revision of that patch with a more reasonable parsing.